[Form control refresh] Submit buttons should not use bold text on iOS

lilyspiniolas opened the pull request:
#### c9ff2b60570e45499e67a9cf5d72a7b567850a1f
<pre>
[Form control refresh] Submit buttons should not use bold text on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=296922">https://bugs.webkit.org/show_bug.cgi?id=296922</a>
<a href="https://rdar.apple.com/157545551">rdar://157545551</a>

Reviewed by Tim Nguyen.

Removed CSS style `font-weight: bold` for submit buttons on iOS.
Rebaselined tests to account for the updated control size.

* LayoutTests/platform/ios-18/editing/selection/3690703-2-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/3690703-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/3690719-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/4397952-expected.txt:
* LayoutTests/platform/ios-18/fast/css/margin-top-bottom-dynamic-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/001-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/blankbuttons-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/button-default-title-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/formmove3-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/input-appearance-height-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/targeted-frame-submission-expected.txt:
* LayoutTests/platform/ios-18/fast/replaced/width100percent-button-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug1318-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug138725-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug39209-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug4429-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug46368-1-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug46368-2-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug7342-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla_expected_failures/bugs/bug1725-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690703-2-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690703-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690719-expected.txt:
* LayoutTests/platform/ios/editing/selection/4397952-expected.txt:
* LayoutTests/platform/ios/fast/css/margin-top-bottom-dynamic-expected.txt:
* LayoutTests/platform/ios/fast/forms/001-expected.txt:
* LayoutTests/platform/ios/fast/forms/blankbuttons-expected.txt:
* LayoutTests/platform/ios/fast/forms/button-default-title-expected.txt:
* LayoutTests/platform/ios/fast/forms/formmove3-expected.txt:
* LayoutTests/platform/ios/fast/forms/input-appearance-height-expected.txt:
* LayoutTests/platform/ios/fast/forms/input-first-letter-expected.txt:
* LayoutTests/platform/ios/fast/forms/targeted-frame-submission-expected.txt:
* LayoutTests/platform/ios/fast/replaced/width100percent-button-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1318-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug138725-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug39209-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug4429-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug46368-1-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug46368-2-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug7342-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug1725-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* Source/WebCore/css/html.css:
(#if defined(WTF_PLATFORM_IOS_FAMILY) &amp;&amp; WTF_PLATFORM_IOS_FAMILY):

Canonical link: <a href="https://commits.webkit.org/299099@main">https://commits.webkit.org/299099@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b32fcbf9412a4250963ca6c36bbbfe1a8602c553

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117829 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37507 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28137 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123967 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69854 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b118726d-f754-4714-b984-b2baba2030b6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119707 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46089 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89419 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59046 "Build is in progress. Recent messages:") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e8a86d48-c891-45f5-a1e9-e82a792391e9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120781 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30413 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105648 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69912 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/441214eb-70c4-416f-99e5-23bd7151e97f) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29477 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23764 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67630 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99833 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/23943 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127056 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44732 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33678 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98083 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45092 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101873 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97871 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24901 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43258 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21228 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41127 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44604 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50278 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44063 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47409 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45753 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -847,6 +847,9 @@ button[type="submit"], button[type="submit"]:active,
     color: white;
+}
+
+button[type="menu"], button[type="menu"]:active {
```
You don't need `, button[type="menu"]:active` here, I don't think anything overrides the font-weight?

Presumably, more of this file could be cleaned up too, with unnecessary `:active` overrides (would be fine to do it separately)

pxlcoder commented on Source/WebCore/css/html.css:
```diff
@@ -847,6 +847,9 @@ button[type="submit"], button[type="submit"]:active,
 button[type="menu"], button[type="menu"]:active {
```
Is there a reason to not make the change for `button[type="menu"]`?

pxlcoder reviewed (commented):
Do we need to rebaseline tests?

lilyspiniolas committed:
[Form control refresh] Submit buttons should not use bold text on iOS
https://bugs.webkit.org/show_bug.cgi?id=296922
rdar://157545551

Reviewed by Tim Nguyen.

Removed CSS style `font-weight: bold` for submit buttons on iOS.
Rebaselined tests to account for the updated control size.

* LayoutTests/platform/ios-18/editing/selection/3690703-2-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/3690703-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/3690719-expected.txt:
* LayoutTests/platform/ios-18/editing/selection/4397952-expected.txt:
* LayoutTests/platform/ios-18/fast/css/margin-top-bottom-dynamic-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/001-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/blankbuttons-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/button-default-title-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/formmove3-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/input-appearance-height-expected.txt:
* LayoutTests/platform/ios-18/fast/forms/targeted-frame-submission-expected.txt:
* LayoutTests/platform/ios-18/fast/replaced/width100percent-button-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug1318-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug138725-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug39209-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug4429-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug46368-1-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug46368-2-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla/bugs/bug7342-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla_expected_failures/bugs/bug1725-expected.txt:
* LayoutTests/platform/ios-18/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690703-2-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690703-expected.txt:
* LayoutTests/platform/ios/editing/selection/3690719-expected.txt:
* LayoutTests/platform/ios/editing/selection/4397952-expected.txt:
* LayoutTests/platform/ios/fast/css/margin-top-bottom-dynamic-expected.txt:
* LayoutTests/platform/ios/fast/forms/001-expected.txt:
* LayoutTests/platform/ios/fast/forms/blankbuttons-expected.txt:
* LayoutTests/platform/ios/fast/forms/button-default-title-expected.txt:
* LayoutTests/platform/ios/fast/forms/formmove3-expected.txt:
* LayoutTests/platform/ios/fast/forms/input-appearance-height-expected.txt:
* LayoutTests/platform/ios/fast/forms/input-first-letter-expected.txt:
* LayoutTests/platform/ios/fast/forms/targeted-frame-submission-expected.txt:
* LayoutTests/platform/ios/fast/replaced/width100percent-button-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1188-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug1318-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug138725-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug2479-3-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug2479-4-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug33855-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug39209-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug4429-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug46368-1-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug46368-2-expected.txt:
* LayoutTests/platform/ios/tables/mozilla/bugs/bug7342-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug1725-expected.txt:
* LayoutTests/platform/ios/tables/mozilla_expected_failures/bugs/bug2479-5-expected.txt:
* Source/WebCore/css/html.css:
(#if defined(WTF_PLATFORM_IOS_FAMILY) && WTF_PLATFORM_IOS_FAMILY):

Canonical link: https://commits.webkit.org/299099@main
